components/pipedrive/sources/common/base.mjs

```diff
@@ -0,0 +1,94 @@
+import pipedriveApp from "../../pipedrive.app.mjs";
+import constants from "../../common/constants.mjs";
+
+export default {
+  props: {
+    pipedriveApp,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      label: "Polling schedule",
+      description: "How often to poll the Pipedrive API",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+  },
+  hooks: {
+    async deploy() {
+      const stream =
+        await this.pipedriveApp.getResourcesStream({
+          resourceFn: this.getResourceFn(),
+          resourceFnArgs: this.getResourceFnArgs(),
+          max: constants.DEFAULT_MAX_ITEMS,
+        });
+
+      await this.processStreamEvents(stream);
+    },
+  },
+  methods: {
+    getLastResourceProperty() {
+      return this.db.get(constants.LAST_RESOURCE_PROPERTY);
+    },
+    setLastResourceProperty(value) {
+      this.db.set(constants.LAST_RESOURCE_PROPERTY, value);
+    },
+    getResourceProperty() {
+      throw new Error("getResourceProperty Not implemented");
+    },
+    getEventObject() {
+      throw new Error("getEventObject Not implemented");
+    },
+    getEventAction() {
+      throw new Error("getEventAction Not implemented");
+    },
+    getResourceFn() {
+      throw new Error("getResourceFn Not implemented");
+    },
+    getResourceFnArgs() {
+      throw new Error("getResourceFnArgs Not implemented");
+    },
+    generateMeta() {
+      throw new Error("generateMeta Not implemented");
+    },
+    done() {
+      throw new Error("done Not implemented");
+    },
```

Comment:
Since `done` is the same in all the sources, we could put it in `base.mjs`.
```suggestion
    done({
      resource, lastResourceProperty,
    }) {
      const property = this.getResourceProperty();
      return lastResourceProperty === String(resource[property]);
    },
```